File: app/src/main/java/com/odnovolov/randomuser/network/AllQueriesWithPrettyFormat.java
package com.odnovolov.randomuser.network;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import javax.inject.Inject;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;


public class AllQueriesWithPrettyFormat implements Interceptor {
    @Inject
    public AllQueriesWithPrettyFormat() {
    }

    @NotNull
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request originalRequest = chain.request();

        HttpUrl originalHttpUrl = originalRequest.url();

        HttpUrl newHttpUrl = originalHttpUrl.newBuilder()
                .addQueryParameter("format", "pretty")
                .build();

        Request newRequest = originalRequest.newBuilder()
                .url(newHttpUrl)
                .build();

        return chain.proceed(newRequest);
    }
}

File: app/src/main/java/com/odnovolov/randomuser/network/AllQueriesWithPrettyFormat.kt
package com.odnovolov.randomuser.network

import okhttp3.HttpUrl
import okhttp3.Interceptor
import okhttp3.Interceptor.Chain
import okhttp3.Request
import okhttp3.Response
import javax.inject.Inject

class AllQueriesWithPrettyFormat @Inject constructor() : Interceptor {
    override fun intercept(chain: Chain): Response {
        val originalRequest: Request = chain.request()

        val originalHttpUrl: HttpUrl = originalRequest.url()

        val newHttpUrl: HttpUrl = originalHttpUrl.newBuilder()
            .addQueryParameter("format", "pretty")
            .build()

        val newRequest: Request = originalRequest.newBuilder()
            .url(newHttpUrl)
            .build()

        return chain.proceed(newRequest)
    }
}